coordinator/clients/smart-contract-client/src/main/kotlin/net/consensys/linea/contract/Web3JContractAsyncHelper.kt

```diff
@@ -232,17 +330,32 @@ class Web3JContractAsyncHelper(
   }
 
   fun executeEthCall(function: Function): SafeFuture<String?> {
-    val gasLimit = gasProvider.getGasLimit(function.name)
-    val tx = Transaction.createFunctionCallTransaction(
-      transactionManager.fromAddress,
-      null,
-      null,
-      gasLimit,
-      contractAddress,
-      FunctionEncoder.encode(function)
-    )
+    return getGasLimit(function)
```

Comment:
This doesn't quite makes sense because `getGasLimit` already does what `executeEthCall` is supposed to do if `useEthEstimateGas` is enabled

Reply:
Yes, I was thinking the same but indeed it's a bit different, if `useEtheEstimateGas` is enabled, we would call `eth_estimateGas` to estimate the gas limit and then explicitly set the it as the `gas` field in `eth_call` to double check against if the "estimated gas limit" is sufficient enough, given that "estimated gas limit" is not a guarantee and could sometimes be insufficient when being used in `eth_call` or `eth_sendRawTransaction`

> In practice, this mismatch can happen, especially in complex transactions or when dealing with contracts that have unpredictable gas usage. To mitigate this:
> - Test with eth_call: Before relying on the estimate for a real transaction, simulate the call with eth_call using the estimated gas to see if it holds up.

wdyt?

Reply:
To remove the extra call of `eth_estimateGas`, I'd changed the aggregation submission flow as:
`eth_estimateGas -> eth_call -> eth_sendRawTransaction`